Change default max partition bytes in DR consumers

ballard26 opened the pull request:
Prior to this PR the default max partition bytes was 1MiB. The default min fetch bytes was 5MiB. And the max wait ms was 500ms.

In cases where only one partition on the shadow cluster shard was being produced to these defaults limited our consumer throughput to 2MiB/s. This is since we'd only be able to read 1MiB every 500ms from the one partition with new data. And in cases where the throughput to that partition exceeded 2MiB/s we'd see lag increase indefinitely between the source and shadow cluster because of this.

Hence this commit sets max partition bytes to equal min fetch bytes to prevent this issue.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.3.x
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes
* none
<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

ballard26 committed:
cluster_link/model: change default max partition bytes

Prior to this commit the default max partition bytes was 1MiB. The
default min fetch bytes was 5MiB. And the max wait ms was 500ms.

In cases where only one partition on the shard was being produced to
these defaults limited our consumer throughput to 2MiB/s. This is since
we'd only be able to read 1MiB every 500ms from the one partition with
new data.

Hence this commit sets max partition bytes to equal min fetch bytes to
prevent this issue.

ballard26 commented:
Another option here is lowering the fetch min bytes to 1MiB. Either would prevent the issue from occurring.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75796-->
<!--545e62a0-41c0-4f55-b464-8bab5f6f2d1c-->
<!--392bfc84-b3dd-45f9-ab8e-566d72486c42-->
<!--title-->
## Retry command for [Build#75796](https://buildkite.com/redpanda/redpanda/builds/75796)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkBasicTests.test_create_default_link
```
<!--/body-->

ballard26 commented on tests/rptest/clients/admin/proto/redpanda/core/admin/v2/cluster_pb2.pyi:
```diff
@@ -34,7 +34,7 @@ class ListKafkaConnectionsRequest(google.protobuf.message.Message):
     ORDER_BY_FIELD_NUMBER: builtins.int
     page_size: builtins.int
-    'The maximum number of connections to return. If unspecified or 0, a\n    default value may be applied. Note that paging is currently not fully\n    supported, and this field only acts as a limit for the first page of data\n    returned. Subsequent pages of data cannot be requested.\n    '
+    'The maximum number of connections to return. If unspecified or 0, a\n    default value may be applied. The server may return fewer connections\n    than requested due to memory constraints; the limit is set to allow\n    listing all connections for a single broker. Consider filtering by\n    node_id to view connections for specific brokers. Note that paging is\n    currently not fully suppor ...
```
I'm guessing someone forgot to run `task rp:generate-ducktape-protos` after modifying the cluster proto at some point. Happy to separate this into a separate commit if folks want.

ballard26 replied:
Should we add a note to `proto/redpanda/README.md` about running the command after updating a proto?

rockwotj replied:
I will just add a github action to check in another PR

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75806-->
<!--86ecd7aa-55b5-44fd-b7a8-f741542a4bcb-->
<!--title-->
## Retry command for [Build#75806](https://buildkite.com/redpanda/redpanda/builds/75806)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkingMetricsTests.test_link_metrics
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--972e1514-6361-4fb3-afe3-423ee1a37558-->
<!--TEST RESULTS-->
<!--857b9dba-741b-47b2-afe9-834c9588a189-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75806">build#75806</a></summary>


| test_class                    | test_method            | test_arguments                                                                    | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                       |
|:------------------------------|:-----------------------|:----------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingMetricsTests     | test_link_metrics      | null                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/75806#019a5c86-e28c-43a9-af83-d3aab349fc6f | FAIL          | 0/21     | The test has failed across all retries                                                                                                                                  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingMetricsTests&test_method=test_link_metrics          |
| ShadowLinkingReplicationTests | test_replication_basic | {"shuffle_leadership": true, "source_cluster_spec": {"cluster_type": "redpanda"}} | integration | https://buildkite.com/redpanda/redpanda/builds/75806#019a5c8a-faff-4ae0-a7dc-22d63efd7c4a | FLAKY         | 17/21    | upstream reliability is '95.91836734693877'. current run reliability is '80.95238095238095'. drift is 14.96599 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_basic |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75995">build#75995</a></summary>


| test_class                       | test_method                | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                              |
|:---------------------------------|:---------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| WriteCachingFailureInjectionTest | test_unavoidable_data_loss | null             | integration | https://buildkite.com/redpanda/redpanda/builds/75995#019a7236-c4a7-4ded-9abf-b932222eca27 | FLAKY         | 19/21    | upstream reliability is '94.76510067114094'. current run reliability is '90.47619047619048'. drift is 4.28891 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=WriteCachingFailureInjectionTest&test_method=test_unavoidable_data_loss |
</details>
<!--/body-->

ballard26 committed:
rptest: increase message count in metrics test

The metrics test checks for a non-zero value for lag. Whether the metric
is non-zero by the time the check occurs depends on whether all messages
have been produced and if the shadow cluster has caught up. This may not
always be the case depending how the test is ran(i.e, in CDT things will
run much faster due to the greater resources).

Before the default max partition bytes was 5MiB in the direct consumer
the shadow cluster could at most consume 1MiB/s from the single
partition topic in the test. This meant that more likely than not the
shadow cluster would still be catching up by the time the metric is
checked. However, not that the default max partition bytes is 5MiB the
throughput limit is no longer the case and the shadow cluster is more
likely than not to be caught up by the time the metric is checked.

This commit increases the messages being produced to be much higher than
before to ensure that the shadow cluster is still catching up by the
time the lag metric is checked. This change also increases the test
runtime from 1min to 2min.
